src/overlays/actors/ovl_En_Attack_Niw/z_en_attack_niw.c

```diff
@@ -27,23 +36,372 @@ const ActorInit En_Attack_Niw_InitVars = {
+    Math_SmoothScaleMaxF(&this->actor.speedXZ, this->unk_2E0, 0.9f, 1.0f);
+    if ((this->actor.gravity == -2.0f) && (this->unk_262 == 0) &&
+        ((this->actor.bgCheckFlags & 8) || (this->unk_25C == 0))) {
+        this->unk_2E0 = 0.0f;
+        this->actor.gravity = 0.0f;
+        this->unk_2DC = 0.0f;
+        this->unk_2D0 = this->actor.posRot.rot.x - 5000.0f;
+        this->actionFunc = func_809B5C18;
+    } else if (this->actor.bgCheckFlags & 1) {
+        func_809B5268(this, globalCtx, 5);
+    } else {
+        func_809B5268(this, globalCtx, 2);
+    }
+}
+
+void func_809B5C18(EnAttackNiw* this, GlobalContext* globalCtx) {
+    if (!func_809B55EC(this, globalCtx)) {
+        Actor_Kill(&this->actor);
+    } else {
+        Math_SmoothScaleMaxMinS(&this->actor.posRot.rot.x, this->unk_2D0, 5, this->unk_2DC, 0);
+        Math_SmoothScaleMaxF(&this->unk_2DC, 5000.0f, 1.0f, 100.0f);
+        Math_SmoothScaleMaxF(&this->actor.velocity.y, 5.0f, 0.3f, 1.0f);
+        func_809B5268(this, globalCtx, 2);
+    }
+}
+
+void EnAttackNiw_Update(Actor* thisx, GlobalContext* globalCtx) {
+    f32 tmpf1;
+    EnAttackNiw* this = THIS;
+    EnNiw* cucco;
+    Player* player = PLAYER;
+    s32 pad;
+    Vec3f sp30;
+    GlobalContext* globalCtx2 = globalCtx;
+
+    this->unk_28C++;
+    if (this->unk_254 != 0) {
+        this->unk_254--;
+    }
+    if (this->unk_258 != 0) {
+        this->unk_258--;
+    }
+    if (this->unk_25A != 0) {
+        this->unk_25A--;
+    }
+    if (this->unk_25E != 0) {
+        this->unk_25E--;
+    }
+    if (this->unk_260 != 0) {
+        this->unk_260--;
+    }
+    if (this->unk_25C != 0) {
+        this->unk_25C--;
+    }
+    if (this->unk_262 != 0) {
+        this->unk_262--;
+    }
+
+    this->actor.shape.rot = this->actor.posRot.rot;
+    this->actor.shape.unk_10 = 15.0f;
+    this->actionFunc(this, globalCtx2);
+    func_8002E4B4(globalCtx, &this->actor, 20.0f, 20.0f, 60.0f, 0x1D);
+
+    if (this->actionFunc == func_809B5670) {
+        func_8002D97C(&this->actor);
+    } else {
+        Actor_MoveForward(&this->actor);
+    }
+
+    if (this->actor.groundY <= -32000.0f) {
+        Actor_Kill(&this->actor);
+        return;
+    }
+
+    if ((this->actor.bgCheckFlags & 0x20) && (this->actionFunc != func_809B5C18)) {
+        Math_Vec3f_Copy(&sp30, &this->actor.posRot.pos);
+        sp30.y += this->actor.waterY;
+        func_8002949C(globalCtx, &sp30, 0, 0, 0, 0x190);
+        this->unk_2DC = 0.0f;
+        this->actor.gravity = 0.0f;
+        this->unk_2E0 = 0.0f;
+        this->unk_2D0 = this->actor.posRot.rot.x - 5000.0f;
+        this->actionFunc = func_809B5C18;
+        return;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Attack_Niw/func_809B59B0.s")
+    tmpf1 = 20.0f;
+    if (this->actor.xyzDistFromLinkSq < SQ(tmpf1)) {
+        cucco = (EnNiw*)this->actor.attachedA;
+        if ((this->actor.attachedA->update != NULL) && (this->actor.attachedA != NULL) && (cucco != NULL) &&
```

Comment:
It seems like `(this->actor.attachedA != NULL) && (cucco != NULL)` are redundant. Have you tried removing one?

Reply:
Yes, I tried playing with the order before when Fig mentioned it, and it has to be there. The asm specifically has it there twice.